pkg/tfbridge/info.go

```diff
@@ -34,6 +34,26 @@ import (
 	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/logging"
 )
 
+type urnCtxKeyType string
+
+const urnCtxKey urnCtxKeyType = "urn"
+
+// WithUrn returns a copy of ctx with the resource URN value
+func WithUrn(ctx context.Context, urn resource.URN) context.Context {
+	return context.WithValue(ctx, urnCtxKey, urn)
+}
+
+// GetUrn gets a resource URN from context
+//
+//	urn, ok := GetUrn(ctx)
+//	if !ok {
+//	    // urn was not found
+//	}
+func GetUrn(ctx context.Context) (resource.URN, bool) {
+	urn, ok := ctx.Value(urnCtxKey).(resource.URN)
+	return urn, ok
+}
```

Comment:
It feels like for any call here, there either should or should not be a result for `GetUrn`, and which one should be knowable statically. Given that, I would recommend an API that doesn't support a failure mode:

```suggestion
// GetUrn gets a resource URN from context.
//
//	urn := GetUrn(ctx)
//
// GetUrn is available on any context associated with a resource. 
// Calling GetUrn on a context associated with an Invoke will panic.
func GetUrn(ctx context.Context) resource.URN {
	urn, ok := ctx.Value(urnCtxKey).(resource.URN)
	return urn, ok
}
```